siatune/ray/tuner.py

```diff
@@ -0,0 +1,96 @@
+# Copyright (c) SI-Analytics. All rights reserved.
+import copy
+import os.path as osp
+
+from ray.air.config import RunConfig
+from ray.tune.tune_config import TuneConfig
+from ray.tune.tuner import Tuner as RayTuner
+
+from siatune.ray import (build_callback, build_scheduler, build_searcher,
+                         build_space, build_stopper)
+
+
+class Tuner:
+    """Wrapper class of :class:`ray.tune.tuner.Tuner`.
+
+    Args:
+        trainable (Callable):
+        work_dir (str):
+        param_space (dict, optional):
+        tune_cfg (dict, optional):
+            Refer to https://github.com/ray-project/ray/blob/ray-2.1.0/python/ray/tune/tune_config.py for details.  # noqa
+        searcher (dict, optional):
+        trial_scheduler (dict, optional):
+        stopper (dict, optional):
+        callbacks (list, optional):
+    """
+
+    def __init__(
+        self,
+        trainable,
+        work_dir,
+        param_space=None,
+        tune_cfg=None,
+        searcher=None,
+        trial_scheduler=None,
+        stopper=None,
+        callbacks=None,
+    ):
+        work_dir = osp.abspath(work_dir)
+
+        if param_space is not None:
+            param_space = build_space(param_space)
+
+        tune_cfg = copy.deepcopy(tune_cfg or dict())
+
+        if searcher is not None:
+            searcher = build_searcher(searcher)
+
+        if trial_scheduler is not None:
+            trial_scheduler = build_scheduler(trial_scheduler)
+
+        if stopper is not None:
+            stopper = build_stopper(stopper)
+
+        if callbacks is not None:
+            if isinstance(callbacks, dict):
+                callbacks = [callbacks]
+            callbacks = [build_callback(callback) for callback in callbacks]
+
+        self.tuner = RayTuner(
+            trainable,
+            param_space=dict(train_loop_config=param_space),
+            tune_config=TuneConfig(
+                search_alg=searcher, scheduler=trial_scheduler, **tune_cfg),
+            run_config=RunConfig(
+                local_dir=work_dir,
+                stop=stopper,
+                callbacks=callbacks,
+                failure_config=None,  # todo
+                sync_config=None,  # todo
+                checkpoint_config=None,  # todo
```

Comment:
Are these required to be edited in this PR? or just future works?

Reply:
I'm thinking